source/drivers/LSM303Accelerometer.cpp

```diff
@@ -237,9 +237,8 @@ int LSM303Accelerometer::setSleep(bool doSleep)
     
     if (!doSleep && (status & LSM303_A_STATUS_SLEEPING))
```

Comment:
Maybe we don't want to wait for a valid sample here, in which case we could just do nothing, and the SLEEPING flag isn't needed.

Reply:
Yeah, that is a bit weird. There was a reason why it was there. Probably to ensure valid data after the driver wakes up and the user has requested data, and we need to ensure the data we return is not 6 months old.

we perhaps shouldn't call requestUpdate() from idleCallback() while the SLEEPING flag is still set though? That could differentiate a specific user request for data from precaching?